diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4643eb3e..ddf3aa7f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -157,9 +157,8 @@ else()
 endif()
 
 # SQL
-find_path(SQLITE3_INCLUDE_DIR NAMES sqlite3.h REQUIRED)
-find_library(SQLITE3_LIBRARY NAMES sqlite3)
-include_directories(${SQLITE3_INCLUDE_DIR})
+include("FindSQLite3.cmake") #find_package(SQLite3 REQUIRED)
+include_directories(${SQLite3_INCLUDE_DIRS})
 
 # LLVM
 find_package(llvm-core 12 REQUIRED)
diff --git a/cmake/phasar_macros.cmake b/cmake/phasar_macros.cmake
index 5b330e5c..d39dea8e 100644
--- a/cmake/phasar_macros.cmake
+++ b/cmake/phasar_macros.cmake
@@ -27,7 +27,7 @@ function(add_phasar_unittest test_name)
     # ${PHASAR_PLUGINS_LIB}
     phasar_pointer
     phasar_typehierarchy
-    ${SQLITE3_LIBRARY}
+    ${SQLite3_LIBRARY_LIST}
     ${Boost_LIBRARIES}
     ${CMAKE_DL_LIBS}
     ${CMAKE_THREAD_LIBS_INIT}
diff --git a/lib/DB/CMakeLists.txt b/lib/DB/CMakeLists.txt
index eb2a1d9e..519b13ec 100644
--- a/lib/DB/CMakeLists.txt
+++ b/lib/DB/CMakeLists.txt
@@ -1,7 +1,7 @@
 file(GLOB_RECURSE DB_SRC *.h *.cpp)
 
 include_directories(
-  ${SQLITE3_INCLUDE_DIR}
+  ${SQLite3_LIBRARY_LIST}
 )
 
 set(PHASAR_LINK_LIBS
@@ -44,7 +44,7 @@ endif()
 
 target_link_libraries(phasar_db
   LINK_PUBLIC
-  ${SQLITE3_LIBRARY}
+  ${SQLite3_LIBRARY_LIST}
 )
 
 set_target_properties(phasar_db
diff --git a/tools/phasar-clang/CMakeLists.txt b/tools/phasar-clang/CMakeLists.txt
index dc065f91..cec1ba5f 100644
--- a/tools/phasar-clang/CMakeLists.txt
+++ b/tools/phasar-clang/CMakeLists.txt
@@ -31,7 +31,7 @@ target_link_libraries(phasar-clang
   # ${PHASAR_PLUGINS_LIB}
   phasar_pointer
   phasar_typehierarchy
-  ${SQLITE3_LIBRARY}
+  ${SQLite3_LIBRARY_LIST}
   ${Boost_LIBRARIES}
   ${CMAKE_DL_LIBS}
   ${CMAKE_THREAD_LIBS_INIT}
diff --git a/tools/phasar-llvm/CMakeLists.txt b/tools/phasar-llvm/CMakeLists.txt
index 128fc16d..6ea1acd9 100644
--- a/tools/phasar-llvm/CMakeLists.txt
+++ b/tools/phasar-llvm/CMakeLists.txt
@@ -30,7 +30,7 @@ target_link_libraries(phasar-llvm
   # ${PHASAR_PLUGINS_LIB}
   phasar_pointer
   phasar_typehierarchy
-  ${SQLITE3_LIBRARY}
+  ${SQLite3_LIBRARY_LIST}
   ${Boost_LIBRARIES}
   ${CMAKE_DL_LIBS}
   ${CMAKE_THREAD_LIBS_INIT}
