diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddf3aa7f..31af239b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -160,6 +160,12 @@ endif()
 include("FindSQLite3.cmake") #find_package(SQLite3 REQUIRED)
 include_directories(${SQLite3_INCLUDE_DIRS})
 
+# Curl
+find_package(CURL REQUIRED)
+include_directories("${CURL_INCLUDE_DIRS}")
+link_libraries(curl)
+
 # LLVM
 find_package(llvm-core 12 REQUIRED)
 include_directories(${llvm-core_INCLUDE_DIRS})
diff --git a/cmake/phasar_macros.cmake b/cmake/phasar_macros.cmake
index 5b330e5c..3f678201 100644
--- a/cmake/phasar_macros.cmake
+++ b/cmake/phasar_macros.cmake
@@ -31,7 +31,7 @@ function(add_phasar_unittest test_name)
     ${Boost_LIBRARIES}
     ${CMAKE_DL_LIBS}
     ${CMAKE_THREAD_LIBS_INIT}
-    curl
+    
     gtest
   )
 
diff --git a/lib/Controller/CMakeLists.txt b/lib/Controller/CMakeLists.txt
index 8e2b9735..6802b6e2 100644
--- a/lib/Controller/CMakeLists.txt
+++ b/lib/Controller/CMakeLists.txt
@@ -33,7 +33,7 @@ endif()
 find_package(Boost COMPONENTS log REQUIRED)
 target_link_libraries(phasar_controller
   LINK_PUBLIC
-  curl
+  
   ${Boost_LIBRARIES}
 )
 
